drivers/include/sps30.h

```diff
@@ -288,7 +288,7 @@ int sps30_read_serial_number(const sps30_t *dev, char *str, size_t len);
 int sps30_reset(const sps30_t *dev);
 
 /**
- * @brief       Sleep the sensor.
+ * @brief       Put the sensor in sleep mode (can only be done from Idle mode)
```

Comment:
Will `_rx_tx_data` generate such an error?

Reply:
What would you recommend then ? 
I can see three options: 
1. Store modes in which sensor is in inside code and return errors if invalid requests made in this PR itself
2. Implement point 1 in a separate PR
3. Expect user to follow the state diagram (I don't think this would happen/ should be done)

Reply:
Why not

 4.  Call `sps30_stop_measurement()` in `sps30_sleep()`

Reply:
> Will _rx_tx_data generate such an error?

No, it won't because of the sensor respond to the request of sleep command.

> Why not
Call sps30_stop_measurement() in sps30_sleep()

I feel it is better to give the user an error when he/she tries to do something which the sensor does not support rather than executing the command.

Reply:
But if I understand correctly, there is no error when the device is in measuring mode and the device will also no go to sleep - is this correct?

Reply:
Yes, you are correct. Practically, the device will just not respond to the request.

Reply:
But that means it will *not* be asleep and keep consuming energy.

Is there any drawback in sending the `sps30_stop_measurement()` command before `SPS30_CMD_SLEEP`?

Reply:
No, there is no drawback.